datalake/src/main/java/com/sequenceiq/datalake/controller/SdxEventController.java

```diff
@@ -0,0 +1,143 @@
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import java.util.zip.ZipEntry;
+import java.util.zip.ZipOutputStream;
+
+import javax.inject.Inject;
+import javax.ws.rs.core.Response;
+import javax.ws.rs.core.StreamingOutput;
+
+import org.springframework.data.domain.PageRequest;
+import org.springframework.data.domain.Pageable;
+import org.springframework.data.domain.Sort;
+import org.springframework.stereotype.Controller;
+
+import com.sequenceiq.authorization.annotation.CustomPermissionCheck;
+import com.sequenceiq.authorization.utils.EventAuthorizationDto;
+import com.sequenceiq.authorization.utils.EventAuthorizationUtils;
+import com.sequenceiq.cloudbreak.auth.crn.Crn;
+import com.sequenceiq.cloudbreak.auth.crn.CrnResourceDescriptor;
+import com.sequenceiq.cloudbreak.common.json.JsonUtil;
+import com.sequenceiq.cloudbreak.structuredevent.event.StructuredEventType;
+import com.sequenceiq.cloudbreak.structuredevent.event.cdp.CDPStructuredEvent;
+import com.sequenceiq.cloudbreak.structuredevent.rest.endpoint.CDPStructuredEventV1Endpoint;
+import com.sequenceiq.cloudbreak.structuredevent.service.db.CDPStructuredEventDBService;
+import com.sequenceiq.cloudbreak.validation.ValidCrn;
+import com.sequenceiq.datalake.entity.SdxCluster;
+import com.sequenceiq.datalake.repository.SdxClusterRepository;
+import com.sequenceiq.sdx.api.endpoint.SdxEventEndpoint;
+
+@Controller
+public class SdxEventController implements SdxEventEndpoint {
+
+    public static final int DEFAULT_PAGE_SIZE = 100;
+
+    public static final int DEFAULT_STARTING_PAGE = 0;
+
+    @Inject
+    private CDPStructuredEventDBService cdpStructuredEventDBService;
+
+    @Inject
+    private EventAuthorizationUtils eventAuthorizationUtils;
+
+    @Inject
+    private SdxClusterRepository sdxClusterRepository;
+
+    @Inject
+    private CDPStructuredEventV1Endpoint cdpStructuredEventV1Endpoint;
+
+    /**
+     * Retrieves audit events for all datalakes and environments that have been associated with the provided environment CRN.
+     *
+     * @param resourceCrn an environment CRN
+     * @param types       types of structured events to retrieve
+     * @return structured events for all datalakes related to the resource CRN of an environment
+     */
+    @Override
+    @CustomPermissionCheck
+    public List<CDPStructuredEvent> getAuditEvents(String resourceCrn, List<StructuredEventType> types) {
+        PageRequest pageable = PageRequest.of(DEFAULT_STARTING_PAGE, DEFAULT_PAGE_SIZE, Sort.by("timestamp").descending());
+
+        List<String> datalakeCrns = getRelatedDatalakeCrns(resourceCrn);
+        List<CDPStructuredEvent> dlEvents = cdpStructuredEventDBService.getPagedEventsOfResources(types, datalakeCrns, pageable).getContent();
+
+        List<CDPStructuredEvent> environmentEvents = cdpStructuredEventV1Endpoint.getAuditEvents(resourceCrn, types, DEFAULT_STARTING_PAGE, DEFAULT_PAGE_SIZE);
```

Comment:
Another question came into my mind: I am just wondering why do you placed this logic into datalake service instead of environment service? based on the logic you are querying events based on environmentCrn

Reply:
We're querying based on environment CRN, but the events that were returning are all for Data Lake events.

I think it makes sens to say that we're querying for `data lake events`, and filtering on the membership of those data lakes in a certain environment.

Reply:
the response will contain the events of environment too like freeipa created, network created etc, which is not related to datalake

Reply:
The response will events the datalake only.